app/services/CommonTaskListService.scala

```diff
@@ -0,0 +1,109 @@
+
+    val ukPropertyUrl: String = s"${appConfig.propertyFrontendUrl}/$taxYear/uk-property/about/start"
+    val foreignPropertyUrl: String = s"${appConfig.propertyFrontendUrl}/$taxYear/foreign-property/about/start"
+    val ukForeignPropertyUrl: String = s"${appConfig.propertyFrontendUrl}/$taxYear/uk-foreign-property/about/start"
+
+    def result(): Future[Seq[TaskListSection]] = {
+      for {
+        allBusinessDataOpt <- service.getBusinessDetails(nino).map {
+          case Right(businessDetails)  => PrePopulationResponse.fromData(businessDetails)
+          case Left(DataNotFoundError) => PrePopulationResponse.noPrePop
+          case Left(err)               =>
+            logger.warn(s"[CommonTaskListService][result] - An error occurred while checking the user's Property data for pre-pop $err")
+            PrePopulationResponse.noPrePop
+        }
+        allJourneys <- repository.fetchAllJourneysUserTaxYear(taxYear, mtdItId)
+        ukJourneyAnswersOpt = allJourneys.find(_.journey == JourneyName.About)
+        foreignJourneyAnswersOpt = allJourneys.find(_.journey == JourneyName.ForeignPropertySelectCountry)
+        ukAndForeignJourneyAnswersOpt = allJourneys.find(_.journey == JourneyName.UkAndForeignPropertyAbout)
+        taskList = toTaskList(allBusinessDataOpt, ukJourneyAnswersOpt, foreignJourneyAnswersOpt, ukAndForeignJourneyAnswersOpt)
+      } yield taskList
+    }
+
+    def toTaskList(prePopulationResponse: PrePopulationResponse,
+                   ukPropertyJourneyAnswersOpt: Option[JourneyAnswers],
+                   foreignPropertyJourneyAnswersOpt: Option[JourneyAnswers],
+                   ukAndForeignPropertyJourneyAnswersOpt: Option[JourneyAnswers]): Seq[TaskListSection] = {
+      val ukPropertyTask: Option[Seq[TaskListSectionItem]] = getPropertyTasks(
+        ifPropertyExists = prePopulationResponse.hasUkPropertyPrePop,
+        taskTitle = TaskTitle.UkProperty,
+        url = ukPropertyUrl,
+        journeyAnswers = ukPropertyJourneyAnswersOpt
+      )
+      val foreignPropertyTask: Option[Seq[TaskListSectionItem]] = getPropertyTasks(
+        ifPropertyExists = prePopulationResponse.hasForeignPropertyPrePop,
+        taskTitle = TaskTitle.ForeignProperty,
+        url = foreignPropertyUrl,
+        journeyAnswers = foreignPropertyJourneyAnswersOpt
+      )
+      val ukForeignPropertyTask: Option[Seq[TaskListSectionItem]] = getPropertyTasks(
+        ifPropertyExists = prePopulationResponse.hasUkPropertyPrePop && prePopulationResponse.hasForeignPropertyPrePop,
+        taskTitle = TaskTitle.UkForeignProperty,
+        url = ukForeignPropertyUrl,
+        journeyAnswers = ukAndForeignPropertyJourneyAnswersOpt
+      )
+      Seq(
+        TaskListSection(SectionTitle.UkPropertyTitle, ukPropertyTask),
+        TaskListSection(SectionTitle.ForeignPropertyTitle, foreignPropertyTask),
+        TaskListSection(SectionTitle.UkForeignPropertyTitle, ukForeignPropertyTask)
+      )
+    }
```

Comment:
MIght be cleaner to make this a `private def` to the class. So that this `result` function isn't too big and doesn't contain a nested function within it.